### DISCUSSION TITLE
regenerate and reindex

### DISCUSSION THREAD

**Discussion by saket424** [blakeblackshear/frigate#14100] [Ask A Question]
I see the option to regenerate the description on a single image and I see an option the reIndex for all images for the purposes of semantic search
It may be useful to offer a way to regenerate the description for all the images especially when the prompts can be tweaked to elicit better outcomes or if we screwed up and plenty of objects have no descriptions and others do.

Thoughts?

**Comment by NickM-27** [comment]
this is not something that I would think makes sense running through the UI, it is very CPU intensive and makes sense if you just updated and want to make sure the content is searchable but doing it periodically doesn't make sense to me. 

Curious to see what other maintainers think

**Reply by hawkeye217** [reply]
I agree. It's very resource-intensive if running locally. Also, those utilizing remote providers will run up against API quotas and rate limits quickly.

**Reply by saket424** [reply]
I am running the API locally self-hosted inferencing on a 4090 gpu. So API quotas are not an issue at least for me.
If you have re-generate for an individual image, why not have a one button for batch-mode re-generate where the user can consciously and on-demand redo the descriptions fully aware that the process can be resource intensive. I am not suggesting it be done periodically like embeddings maybe but only when explicitly requested. Anyhow, I respect your views if you are opposed to this suggestion. Especially when there is thrashing in the prompt and model, this redo seems useful
